airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/DebeziumRecordPublisher.java

```diff
@@ -86,7 +86,10 @@ public void start(Queue<ChangeEvent<String, String>> queue) {
           // more on the tombstone:
           // https://debezium.io/documentation/reference/configuration/event-flattening.html
           if (e.value() != null) {
-            queue.add(e);
+            boolean inserted = false;
+            while (!inserted) {
+              inserted = queue.offer(e);
```

Comment:
should we do a small `sleep` here to ensure we don't busy-loop inside this thread?

Reply:
@sherifnada What do you think would be a right sleep value?

Reply:
ideally expo backoff for 1/2/4/8/16 ms maybe capping at half a second. But barring that a magic constant (erring on the smaller side) should do. For example 10ms.